extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/KafkaConsumerMonitor.java

```diff
@@ -60,18 +131,33 @@ public boolean doMonitor(final ServiceEmitter emitter)
   {
     for (final Map.Entry<MetricName, ? extends Metric> entry : consumer.metrics().entrySet()) {
       final MetricName metricName = entry.getKey();
+      final KafkaConsumerMetric kafkaConsumerMetric = METRICS.get(metricName.name());
+
+      if (kafkaConsumerMetric != null &&
+          kafkaConsumerMetric.getDimensions().equals(metricName.tags().keySet())) {
```

Comment:
Instead of requiring both the sets to be equal, should we just check for a contains relationship?

```java
kafkaConsumerMetric != null &&
   metricName.tags().keySet().containsAll(kafkaConsumerMetric.getDimensions())
```

Reply:
Please also add a comment on why we need to check for the dimensions. The older code had this comment:

```java
    // Certain metrics are emitted both as grand totals and broken down by topic; we want to ignore the grand total and
    // only look at the per-topic metrics. See https://kafka.apache.org/documentation/#consumer_fetch_monitoring.
```

Reply:
Oops, yes, that comment is the reason why we need to look for equal sets. I have restored it.